tests/test_rpki_parser.py

```diff
@@ -96,3 +96,48 @@ def test_roa_incompletes(tmp_path, capsys):
     assert len(final_lines) == 0, "No rows should be written"
     captured = capsys.readouterr()
     assert "Incompletes: 2" in captured.out
+
+
+def test_roa_valid_until_fallback(tmp_path):
+    '''Test ROA selection falls back to later valid_until'''
+    epoch = "111111111"
+    context = create_test_context(tmp_path, epoch)
+    setup_test_data(context)
+    parse_rpki(context)
+
+    final_path = os.path.join(context.out_dir_rpki, "rpki_final.txt")
+    with open(final_path, "r") as f:
+        entries = [line.strip() for line in f.readlines()]
```

Comment:
Nit: Quite a bit of dublication between the newly added tests. Maybe you can extract this into a helper function that is local in this file?

Reply:
Meh, I will just merge this since it looks fine otherwise. Maybe keep this in mind for the next time you add something to this file :)